Framework/jmaqs-webservices/src/main/java/com/magenic/jmaqs/webservices/WebServiceDriverManager.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019 (C) Magenic, All rights Reserved
+ */
+
+package com.magenic.jmaqs.webservices;
+
+import com.magenic.jmaqs.base.BaseTestObject;
+import com.magenic.jmaqs.base.DriverManager;
+
+import java.net.URISyntaxException;
+import java.util.function.Supplier;
+import org.apache.http.impl.client.CloseableHttpClient;
+
+/**
+ * Web Service Driver Manager Class
+ */
+public class WebServiceDriverManager extends DriverManager {
+  /**
+   * Web Service Driver variable
+   */
+  private WebServiceDriver webServiceDriver;
+
+  /**
+   * Instantiates a new Web Service Driver Manager.
+   *
+   * @param getDriverFunction Function that specifies how to get the driver.
+   * @param baseTestObject    The Base Test Object.
+   */
+  public WebServiceDriverManager(Supplier<CloseableHttpClient> getDriverFunction,
+      BaseTestObject baseTestObject) {
+    super(getDriverFunction, baseTestObject);
+    this.webServiceDriver = new WebServiceDriver(getDriverFunction.get());
+  }
+
+  /**
+   * Instantiates a new Web Service Driver Manager.
+   *
+   * @param driver            Web Service Driver
+   * @param baseTestObject    The Base Test Object.
+   */
+  public WebServiceDriverManager(WebServiceDriver driver, BaseTestObject baseTestObject) {
+    super(() -> driver, baseTestObject);
+    this.webServiceDriver = driver;
+  }
+
+  /**
+   * Get the Web Service Driver.
+   *
+   * @return  The Web Service Driver.
+   */
+  public WebServiceDriver getWebServiceDriver() throws URISyntaxException {
+    // Create default Web Service Driver and instantiate Base HTTP Client.
+    if (this.webServiceDriver == null) {
+      this.webServiceDriver = new WebServiceDriver(WebServiceConfig.getWebServiceUri());
+      this.webServiceDriver.setHttpClient(this.webServiceDriver.getHttpClient(MediaType.APP_JSON.getMediaTypeString()));
+    }
```

Comment:
Would like feedback on this part of the code. This process looks too clunky but it works?

Reply:
since its just going to be in JDK 8 and it works, it seems good to me.

Reply:
There is a smell to it, but I can't put my finger on what it is at the moment.

Reply:
To me, it is mostly the syntax/naming that has the smell to it. The GetHttpClient Method is what actually instantiates the default HttpClient within WebServiceDriver. In this case, we wouldn't even need to use the setHttpClient as it's already instantiated when calling getHttpClient. Realistically, we don't even need to set the HttpClient.

Reply:
New solution created to handle both a WebServiceDriver type supplier and CloseableHttpClient Supplier.